flocker/node/agents/cinder.py

```diff
@@ -515,9 +529,18 @@ def _blockdevicevolume_from_cinder_volume(cinder_volume):
     )
 
 
-@auto_openstack_logging(ICinderVolumeManager, "cinder_client.volumes")
-@auto_openstack_logging(INovaVolumeManager, "nova_client.volumes")
-@auto_openstack_logging(INovaServerManager, "nova_client.servers")
+@auto_openstack_logging(ICinderVolumeManager, "_cinder_volumes")
+class _LoggingCinderVolumeManager(PRecord):
+    _cinder_volumes = field(mandatory=True)
+
+@auto_openstack_logging(INovaVolumeManager, "_nova_volumes")
+class _LoggingNovaVolumeManager(PRecord):
+    _nova_volumes = field(mandatory=True)
+
+@auto_openstack_logging(INovaServerManager, "_nova_servers")
+class _LoggingNovaServerManager(PRecord):
+    _nova_servers = field(mandatory=True)
+
 def cinder_api(cinder_client, nova_client, cluster_id):
```

Comment:
Looks like `auto_openstack_logging` (or anything defined with `interface_decorator`) silently accepts functions and does nonsense with them - but in a way that doesn't immediately obviously break things. :/  Probably worth filing an issue for making that fail in an obvious way instead.